### PR TITLE
refactor(scheduler): Move priority types to priorities package

### LINKED ISSUE #71865: [scheduler cleanup phase 2]: Move priority types from `pkg/scheduler/algorithm` to `pkg/scheduler/algorithm/priorities`
**PLEASE COORDINATE WITH SIG-SCHEDULING BEFORE WORKING ON THIS ISSUE.** We want to avoid the situation where two people accidentally start work on the same fix.

Main tracking issue: https://github.com/kubernetes/kubernetes/issues/68951

If you contributed in phase 1, please give others a chance before claiming this issue. You can also help out by reviewing the phase 2 PRs.

The priority types in `pkg/scheduler/algorithm` (`PriorityConfig`, `PriorityMetadata`, etc.) should be moved to the package that implements these types, `pkg/scheduler/algorithm/priorities`. This is a relatively minor change, but having all the priority stuff in one package will make it easier to refactor in future cleanup.

Note: `pkg/scheduler/algorithm` may not import `pkg/scheduler/algorithm/priorities` as that would cause an import loop.

/sig scheduling
/kind cleanup
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by sonasingh46**
**What this PR does / why we need it**:
This moves the priority types from the algorithm package
to priorities package.
Idea is to move the type to the packages where it is
implemented. This will ease the future refactor process.

**What type of PR is this?**
/kind cleanup
/sig scheduling

**Which issue(s) this PR fixes**:
This fixes #71865 

**Does this PR introduce a user-facing change?**:
 "NONE"

### CODE REVIEW

File: pkg/scheduler/algorithm/priorities/types.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*Copyright 2016 The Kubernetes Authors.
```
Comment (wgliang) [inline pkg/scheduler/algorithm/priorities/types.go]: 2016-> 2018

Comment (sonasingh46) [reply pkg/scheduler/algorithm/priorities/types.go]: Done

File: pkg/scheduler/algorithm/priorities/types_test.go [hunk 2]
```diff
@@ -14,14 +14,14 @@ See the License for the specific language governing permissions and
...
+package priorities
 
 import (
-	"testing"
-
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm"
 	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+	"testing"
```
Comment (denkensk) [inline pkg/scheduler/algorithm/priorities/types_test.go]: move "testing" to top

**Review by misterikkit — CHANGES_REQUESTED** [review]
You will need to update all the call sites to use the new import path, and then run `./hack/update-bazel.sh` to fix the BUILD files.

**Review by misterikkit — CHANGES_REQUESTED** [review]
lgtm overall

one minor comment about import alias

please squash commits and then let's get it approved and merged.

(People might not be around next week to approve, but who knows?)

File: pkg/scheduler/core/extender_test.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/kubernetes/pkg/scheduler/algorithm"
+	algorithmpriorities "k8s.io/kubernetes/pkg/scheduler/algorithm/priorities"
```
[the remark this answers is not in the document]
Comment (sonasingh46) [reply pkg/scheduler/core/extender_test.go]: fixed

**Review by misterikkit — APPROVED** [review]
/lgtm

Let's get this rebased and approved! :confetti_ball:

**Review by misterikkit — APPROVED** [review]
/lgtm
/assign @bsalamat @k82cn for approval

**Review by ravisantoshgudimetla — APPROVED** [review]
Thanks @sonasingh46 for the cleanup

/approve

**Pull request merged**